symfit/core/fit.py

```diff
@@ -335,10 +497,18 @@ def __call__(self, *args, **kwargs):
         :return: A namedtuple of all the dependent vars evaluated at the desired point. Will always return a tuple,
             even for scalar valued functions. This is done for consistency.
         """
-        bound_arguments = self.__signature__.bind(*args, **kwargs)
-        Ans = namedtuple('Ans', [var.name for var in self])
-        return Ans(*self.eval_components(**bound_arguments.arguments))
+        Ans = variabletuple('Ans', self)
+        return Ans(*self.eval_components(*args, **kwargs))
+
 
+class BaseGradientModel(BaseCallableModel):
+    """
+    Baseclass for models which have a gradient. Such models are expected to
+    implement an `eval_jacobian` function.
+
+    Any subclass of this baseclass which does not implement its own
+    `eval_jacobian` will inherit a finite difference gradient.
```

Comment:
addition: that should *not* be used during the fit.

Reply:
Then what is it there for? :P This finite difference jacobian is used during the fit, it comes in particularly handy for ODEModels which otherwise would not have a gradient (because I haven't had time/the need to go through the mathematics.)

Reply:
It's currently *not* used for the fit itself, just for the calculation/estimation of the covariance matrix. It's too expensive to calculate it every minimisation step --- doing the minimisation without or with approximated jacobians is faster.

Reply:
Currently it is being used actually. (tested with a simple print) Because if a model has an eval_jacobian, that jacobian is passed on to the minimizer. Or rather, if both the model and the objective have eval_jacobian.

This is indeed new behavior, because before we checked if the model had `numerical_jacobian`. Perhaps we can now change that to checking for `jacobian_model` on the model, or accept that if a model inherits directly from BaseGradientModel, currently only ODEModel, it has a more expensive jacobian.

Because I do think the current decision tree makes more sense than before.

Reply:
It shouldn't ;)
Maybe move finite difference to `BaseModel` or `CallableModel` since we can do it with *any* model? And then check if it's a `JacobianModel` to see if we pass anything to the minimizer/objective.

Reply:
Fit now checkes if a model has `jacobian_model` to determine if it should pass `objective.eval_jacobian` to the minimizer, and so anything inheriting from `BaseGradientModel` is now save in terms of the minimization process, but will still have a method to calculate the covariance matrix. Order has been restored to the universe.